librz/syscall/syscall.c

```diff
@@ -42,22 +42,22 @@ RZ_API void rz_syscall_free(RzSyscall *s) {
 	}
 }
 
-static Sdb *openDatabase(Sdb *db, const char *name) {
+static bool load_sdb(Sdb **db, const char *name) {
+	rz_return_val_if_fail(db, false);
 	char *file = rz_str_newf(RZ_JOIN_3_PATHS("%s", RZ_SDB, "%s.sdb"),
 		rz_sys_prefix(NULL), name);
 	if (rz_file_exists(file)) {
-		if (db) {
-			sdb_reset(db);
-			sdb_open(db, file);
+		if (*db) {
+			sdb_reset(*db);
+			sdb_open(*db, file);
 		} else {
-			db = sdb_new(0, file, 0);
+			*db = sdb_new(0, file, 0);
 		}
-	} else {
-		sdb_free(db);
-		db = sdb_new0();
+		free(file);
```

Comment:
Shouldn't this be the opposite?

Reply:
It should return true if the file exists, right?

Reply:
Yes, I think so.

Reply:
> It should return true if the file exists, right?

The diff is a bit confusing, but that's what the code does.